llama-index-core/llama_index/core/base/llms/types.py

```diff
@@ -46,20 +47,34 @@ class TextBlock(BaseModel):
 
 class ImageBlock(BaseModel):
     block_type: Literal["image"] = "image"
-    image: Optional[str] = None
-    image_path: Optional[str] = None
-    image_url: Optional[str] = None
-    image_mimetype: Optional[str] = None
+    image: bytes | None = None
+    path: FilePath | None = None
+    url: AnyUrl | str | None = None
+    image_mimetype: str | None = None
+
+    @field_validator("image", mode="after")
+    @classmethod
+    def image_to_base64(cls, image: bytes) -> bytes:
+        """Store the image as base64."""
+        return base64.b64encode(image)
```

Comment:
Yes, it will encode it again. The process is safe so `resolve_image` will return `BytesIO` containing the base64 encoded string instead of actual binary data for the image. Depending on the caller, this might still work.